### PR TITLE
Update completion help to use root command in examples

### PR DESCRIPTION
**Pull request by juanvallejo**
Addresses: https://bugzilla.redhat.com/show_bug.cgi?id=1349240

### CODE REVIEW

**Comment by juanvallejo** [comment]
[test]

File: pkg/cmd/cli/cmd/wrappers.go [hunk 1]
```diff
@@ -185,28 +185,34 @@ const (
...
+		cmdHelpString = "openshift admin"
+	} else if strings.HasSuffix(fullName, "completion") {
+		cmdHelpName = "openshift"
+		cmdHelpString = "openshift"
+	}
 
-	completionExample = `  # Generate the openshift cli completion code for bash
-  source <(oc completion bash)
+	cmd := kcmd.NewCmdCompletion(f.Factory, out)
+	cmd.Long = `This command prints shell code which must be evaluated to provide interactive
```
Comment (fabianofranz) [inline pkg/cmd/cli/cmd/wrappers.go]: Use `fmt.Printf` and keep the messages in the `completionLong` and `completionExample` const.

File: pkg/cmd/cli/cmd/wrappers.go [hunk 2]
```diff
@@ -185,28 +185,34 @@ const (
...
   %[1]s exec mypod -c ruby-container -i -t -- bash -il`
+)
 
-	completionLong = `This command prints shell code which must be evaluation to provide interactive
-completion of openshift cli commands.`
+func NewCmdCompletion(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	cmdHelpString := "openshift cli"
+	cmdHelpName := fullName
+
+	if strings.HasSuffix(fullName, "admin") || strings.HasSuffix(fullName, "oadm") {
```
Comment (fabianofranz) [inline pkg/cmd/cli/cmd/wrappers.go]: This logic looks a little confusing... Can't you just use `fullName` directly? I believe it gives you `openshift`, `openshift cli`, `openshift adm`, `oadm` or `oc`, depending on what's being used, and I think it's what we want.

**Comment by juanvallejo** [comment]
@fabianofranz 
> This logic looks a little confusing... Can't you just use fullName directly? I believe it gives you openshift, openshift cli, openshift adm, oadm or oc, depending on what's being used, and I think it's what we want.

Thanks for the feedback on that, https://github.com/openshift/origin/pull/9531/commits/0de480af1b9b348ac51750de971fe8db91dad181 uses fullName directly.

```
if strings.HasSuffix(fullName, "completion") {
  		cmdHelpName = "openshift"
}
```
I still kept the above check in place, as whenever `openshift completion...` is used, it places `completion` in the help and examples twice without it (e.g. `openshift completion completion bash > bash_completion.sh`)

**Comment by juanvallejo** [comment]
@fabianofranz or @kargakis could you take a look please?

File: pkg/cmd/cli/cmd/wrappers.go [hunk 3]
```diff
@@ -178,38 +178,48 @@ func NewCmdCreate(parentName string, f *clientcmd.Factory, out io.Writer) *cobra
...
-  %[1]s exec mypod -c ruby-container -i -t -- bash -il`
-
-	completionLong = `This command prints shell code which must be evaluation to provide interactive
-completion of openshift cli commands.`
-
-	completionExample = `  # Generate the openshift cli completion code for bash
-  source <(oc completion bash)
+	completionExample = `  # Generate the %s completion code for bash
+  %s completion bash > bash_completion.sh
+  source bash_completion.sh
```
Comment (fabianofranz) [inline pkg/cmd/cli/cmd/wrappers.go]: I might be missing something, why not `source <(%s completion bash)`?

Comment (juanvallejo) [inline pkg/cmd/cli/cmd/wrappers.go]: Ah, Xingxing Xia stated in the trello card for this PR [1] that "the behavior `source <(...)` does not work on Mac OS by design" [2]

[1] https://trello.com/c/r3EaiA9F/672-3-cli-oc-completion-command
[2] https://trello.com/c/r3EaiA9F/672-3-cli-oc-completion-command#comment-576ce0bafbadb62268db5234

**Comment by fabianofranz** [comment]
[merge]

**Comment by fabianofranz** [comment]
[merge]

**Comment by fabianofranz** [comment]
flake, re[merge]

**Pull request merged**